### ISSUE TITLE
405 Error

### ISSUE THREAD

**Issue by machineghost** [jaredhanson/passport-google-oauth2#40]
Sometimes when using this library I get 405 errors from Google (and then sometimes it works just fine):

Request URL:https://accounts.google.com/o/oauth2/v2/auth?response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fcallback%2Fgoogle&scope=profile&client_id=637989435204-fivtb7isf82lanreup4qsuhvn3s7hl15.apps.googleusercontent.com
Request Method:OPTIONS
Status Code:405 

Which are then followed by:

Failed to load https://accounts.google.com/o/oauth2/v2/auth?response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fcallback%2Fgoogle&scope=profile&client_id=637989435204-fivtb7isf82lanreup4qsuhvn3s7hl15.apps.googleusercontent.com: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 405. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

But the GitHub page makes no mention of why I'd get XSS errors, or how I can fix them.  It seems like sometimes restarting my server fixes things, and other times it doesn't, but that's as far as I've gotten.

Any assistance would be appreciated.

**Comment by marxys** [comment]
After reading your post, I believe I partially understand what's wrong. This error occur when passport is used in an asynchronous request and then fail to the same origin policy. Have you some javascript which can change form behavior to be able to be send with an XMLHTTPRequest ?

This seems to be a good behavior because it's coud be a security issue... But practically it coud be great to have a solution for asynchronous request. I will exploring this issue deeply tomorrow but if someone have an idea, I m interested !

**Comment by SONGJIHYOEK** [comment]
https://medium.com/@wannabeph/xmlhttprequest-error-when-trying-to-google-login-a86620dbb0aa

**Comment by nishant-dani** [comment]
Was there any workarounds to this issue?  I am having the identical problem on async access.

**Comment by chetanyakan** [comment]
> This would be helpful.
> 
> https://medium.com/@wannabeph/xmlhttprequest-error-when-trying-to-google-login-a86620dbb0aa

TL;DR Use an `Anchor tag` for login redirect instead of making an `Axios/Fetch/XHR` request.
